github/VisualStudio
# [wip] Allow comment margin to appear on editable diff view

### What this PR does

Previously the comment margin would only appear on pull request files that are open in a "full" editor window. This pull request allows the comment margin to also appear on editable diff windows.

### How to test

1. Check `Show PR comments in editor margin` feature flag option
![image](https://user-images.githubusercontent.com/11719160/53354497-c6e1ce00-391e-11e9-904b-404da6b38bb8.png)

2. Open a pull request and checkout its branch
![image](https://user-images.githubusercontent.com/11719160/53354552-eaa51400-391e-11e9-89fd-3af9bbe6902e.png)

3. Right-click on a pull request file and `View Changes in Solution`
![image](https://user-images.githubusercontent.com/11719160/53354640-54252280-391f-11e9-83e1-14359112bb13.png)

4. Toggle the check-box in the margin
![image](https://user-images.githubusercontent.com/11719160/53354701-440d4300-391f-11e9-8e58-9257bcbc07ac.png)

5. Click comment glyph in margin to view comment
![image](https://user-images.githubusercontent.com/11719160/53354744-5a1b0380-391f-11e9-8b93-514d48e246f0.png)

### Observations

This functionality wasn't ready for prime time and has remained behind a feature flag.

#### What didn't work well

- Difficult to keep track of files/comments that are part of pull request
- The comment might no longer be relevant to the code it appears on
- Can't comment on code until it has been pushed

#### What worked well

- Extra space in full editor window to view comments

### Possible future work

- Being able to mark comment threads as resolved
- Map of comments on pull request files
- Map of changed files in pull request
